### ISSUE TITLE
New Rule: detect throw `Nothing`

### ISSUE THREAD

**Issue by zmunm** [detekt/detekt#3600]
## Expected Behavior of the rule

If throw Nothing, clear the throw.
Preconditions.kt
```kotlin
@kotlin.internal.InlineOnly
public inline fun error(message: Any): Nothing = throw IllegalStateException(message.toString())
```

non compliant
```kotlin
throw error("text")
```

compliant
```kotlin
error("text")
```

**Comment by BraisGabin** [comment]
Could you provide more information? Is this a request for a new rule or is this a false positive in a rule?

**Comment by BraisGabin** [comment]
I read about this pattern. I read that the users do this to help the readability of the code. This way you don't need to know what that function does. You know that it will throw an exception. I don't know if I agree with a rule pointing out this cases. But this could be a `@Controversial` style rule.
